### PR TITLE
doc: Adjust RELEASING.md documentation with new process

### PR DESCRIPTION
**Pull request by AgustinBettati**
## Description

Link to any related issue(s): CLOUDP-241730

The diff might be hard to follow, would advise reading [RELEASING.md](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/CLOUDP-241730/RELEASING.md) directly.

The most significant change is that all of the steps related to generating the changelog are no longer needed.

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: RELEASING.md [hunk 1]
```diff
@@ -1,54 +1,30 @@
...
-  - Using an existing tag: Should be left empty (default `false` creates a new tag from `master`). This should be set to `true` only if you want to re-use an existing tag for the release and the tag name should adhere to our version number format.
-
-- You will see the release in the [GitHub Release page](https://github.com/mongodb/terraform-provider-mongodbatlas/releases) once the [release action](.github/workflows/release.yml) has completed.
+### Revise jira release
+Before triggering a release view the corresponding [unreleased jira page](https://jira.mongodb.org/projects/CLOUDP?selectedItem=com.atlassian.jira.jira-projects-plugin:release-page&status=unreleased&contains=terraform) to ensure there are no pending tickets. In case there are pending tickets, verify with the team if the expectation is to have them included within the current release. After release workflow is successful the version will be marked as released automatically.
```
Comment (EspenAlbert) [inline RELEASING.md]:
```suggestion
Before triggering a release, view the corresponding [unreleased jira page](https://jira.mongodb.org/projects/CLOUDP?selectedItem=com.atlassian.jira.jira-projects-plugin:release-page&status=unreleased&contains=terraform) to ensure there are no pending tickets. In case there are pending tickets, verify with the team if the expectation is to have them included within the current release. After release workflow is successful the version will be marked as released automatically.
```

File: RELEASING.md [hunk 2]
```diff
@@ -1,54 +1,30 @@
...
+  - Using an existing tag: Should be left empty (default `false` creates a new tag from `master`). This should be set to `true` only if you want to re-use an existing tag for the release. This can be helpful for rerunning a failed release process in which the tag has already been created.
 
-- Create a new doc in /website/docs/guides/X.Y.0-upgrade-guide.html. This will contain a summary of the most significant features and breaking changes. Additional information that can be helpful to users can be defined here.
+#### How to create a pre-release (not part of regular process)
+Pre-releases are not needed for a regular release process, but they can be generated for exceptional cases (e.g. sharing with external teams for additional testing). The process is the same as a regular release except for the format of the version number which must be `vX.Y.Z-pre` with additional numbers at the end if needed. When a pre release is triggered steps related to updating the changelog header or updating the jira release version are skipped.
```
Comment (EspenAlbert) [inline RELEASING.md]:
```suggestion
Pre-releases are not needed for a regular release process, but they can be generated for exceptional cases (e.g. sharing with external teams for additional testing). The process is the same as a regular release except for the format of the version number which must be `vX.Y.Z-pre` with additional numbers at the end if needed. When a pre release is triggered, steps related to updating the changelog header or updating the jira release version are skipped.
```

File: RELEASING.md [hunk 3]
```diff
@@ -1,54 +1,30 @@
...
 
-### Release the provider
-- Follow the same steps in the pre-release but provide the final release tag (example `v1.9.0`). Harshicorp has a process in place that will retrieve the latest release from the GitHub repository and add the binaries to the Hashicorp Terraform Registry.
+### Post-trigger checks
+- You will see the release in the [GitHub Release page](https://github.com/mongodb/terraform-provider-mongodbatlas/releases) once the [release action](.github/workflows/release.yml) has completed. Harshicorp has a process in place that will retrieve the latest release from the GitHub repository and add the binaries to the Hashicorp Terraform Registry (more details [here](https://developer.hashicorp.com/terraform/registry/providers/publishing#webhooks)).
```
Comment (EspenAlbert) [inline RELEASING.md]: typo: Harshicorp -> Hashicorp

**Review by EspenAlbert — COMMENTED** [review]
Nice! Love how much simpler it is now.
Only minor changes

File: RELEASING.md [hunk 4]
```diff
@@ -1,54 +1,30 @@
...
+### Verify upgrade guide is defined
 
-The PR includes the following changes:
+**Note**: Only applies if the right most version digit is 0 (considered a major or minor version in [semantic versioning](https://semver.org/)).
 
-#### Generate the CHANGELOG.md 
-We use a tool called [github changelog generator](https://github.com/github-changelog-generator/github-changelog-generator) to automatically update our changelog. It provides options for downloading a CLI or using a docker image with interactive mode to update the CHANGELOG.md file locally.
+- A doc in /website/docs/guides/X.Y.0-upgrade-guide.html must be defined containing a summary of the most significant features, breaking changes, and additional information that can be helpful. The expectation is that this file is created during relevant pull requests and not during the release process.
 
-- Update `since_tag` and `future-release` in [.github_changelog_generator](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/.github_changelog_generator)
```
Comment (lantoli) [inline RELEASING.md:36]: can you also delete file .github_changelog_generator ?

**Review by lantoli — APPROVED** [review]

File: RELEASING.md (hunk 4 above)
Comment (AgustinBettati) [reply RELEASING.md:36]: thanks, removed.

File: RELEASING.md (hunk 3 above)
Comment (AgustinBettati) [reply RELEASING.md]: thanks for the catch

**Review by lantoli — APPROVED** [review]

**Review by oarbusi — APPROVED** [review]
